### PR TITLE
[STORM-2071]  Add in retry after rebalance in unit test

### PR DESCRIPTION
**Pull request by ppoulosk**
This retries the command a five times after rebalance, as it was failing on some systems.

### CODE REVIEW

File: storm-core/test/clj/org/apache/storm/nimbus_test.clj [hunk 1]
```diff
@@ -1218,8 +1218,16 @@
               (submit-local-topology nimbus "t1" {} topology)
               ;; Instead of sleeping until topology is scheduled, rebalance topology so mk-assignments is called.
               (.rebalance nimbus "t1" (doto (RebalanceOptions.) (.set_wait_secs 0)))
-              (Thread/sleep 1000)
-              (.deactivate nimbus "t1")
+              ;; One second doesn't work on all systems.  Try it five times
+              (loop [n 5]
+                (if-let [result (try
+                                  [(Thread/sleep 1000)
```
Comment (abellina) [inline storm-core/test/clj/org/apache/storm/nimbus_test.clj]: how about:
(try
  (do
     (Thread/sleep 1000)
     (.deactivate nimbus "t1"))
  (catch ...

File: storm-core/test/clj/org/apache/storm/nimbus_test.clj [hunk 2]
```diff
@@ -1218,8 +1218,16 @@
...
+                                   (.deactivate nimbus "t1")]
+                                  (catch Exception e
+                                    (when (zero? n)
+                                      (throw e))))]
+                         (result 0)
```
Comment (abellina) [inline storm-core/test/clj/org/apache/storm/nimbus_test.clj]: these should be aligned with (if-let +2 spaces from the opening parenthesis.

File: storm-core/test/clj/org/apache/storm/nimbus_test.clj (hunk 1 above)
Comment (HeartSaVioR) [inline storm-core/test/clj/org/apache/storm/nimbus_test.clj]: +1 to this one. Seems clearer.

**Comment by HeartSaVioR** [comment]
+1 once comments from @abellina are addressed. Nice finding.

File: storm-core/test/clj/org/apache/storm/nimbus_test.clj (hunk 1 above)
Comment (ppoulosk) [inline storm-core/test/clj/org/apache/storm/nimbus_test.clj]: Agreed.  Will fix.  Thanks, @abellina.

**Comment by ppoulosk** [comment]
Coded fix, retested, and pushed.  Thanks!!

**Comment by abellina** [comment]
+1 thanks @ppoulosk

**Comment by HeartSaVioR** [comment]
From JIRA comment @revans2 said he fixed this issue from Supervisor V2 patch.

@revans2 Since we saw builds failing (high chance from my dev.) how about addressing build issue before than Supervisor V2?

@ppoulosk @revans2 
Please have a cross look at each patch. We need to choose one which will be merged.

**Comment by revans2** [comment]
I would be happy to pull that fix back into a separate patch.  @ppoulosk this patch looks fine to me, my only concern would be documenting it a little more explaining that the state change might not have completed yet, and until it does we cannot rebalance.  Alternatively if you want to pull in my changes, by replacing sleep with the following I would be OK with that too.

```
(defn wait-for-status [nimbus name status]
  (while-timeout 5000
    (let [topo-summary (first (filter (fn [topo] (= name (.get_name topo))) (.get_topologies (.getClusterInfo nimbus))))
          topo-status (if topo-summary (.get_status topo-summary) "NOT-RUNNING")]
      (log-message "WAITING FOR "name" TO BE " status " CURRENT " topo-status)
      (not= topo-status status))
    (Thread/sleep 100)))
...
(wait-for-status nimbus "t1" "ACTIVE") 
```

Either solution is fine with me.

**Comment by HeartSaVioR** [comment]
@ppoulosk Do you have any update on this?

**Comment by ppoulosk** [comment]
@HeartSaVioR @revans2 I have no preference which solution gets pulled in first.  

I fixed the comment, retested, squashed the three commits into one and pushed it to my branch.

Thanks!

**Comment by HeartSaVioR** [comment]
@ppoulosk Could we close this as Supervisor V2 was merged? Thanks.

**Comment by ppoulosk** [comment]
Yes.  Thanks!

**Pull request closed without merging**